src/AssetManager/Service/AssetFilterManager.php

```diff
@@ -144,8 +151,13 @@ protected function ensureByFilter(AssetInterface $asset, $filter)
                 'No filter found for ' . $filter
             );
         }
-
-        $filterInstance = new $filterClass;
+        
+        if (!array_key_exists($filterClass, $this->filterInstances)) {
```

Comment:
`$this->filterInstances` can only contain instances of `Filter`, therefore `isset()` is sufficient here (instead of `array_key_exists()`)